CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/NotesViewTopComponent.java

```diff
@@ -0,0 +1,176 @@
+@ActionID(
+        category = "Window",
+        id = "au.gov.asd.tac.constellation.views.notes.NotesViewTopComponent")
+@ActionReferences({
+    @ActionReference(path = "Menu/Experimental/Views", position = 500),
+    @ActionReference(path = "Shortcuts", name = "CA-N")})
+@TopComponent.OpenActionRegistration(
+        displayName = "#CTL_NotesViewAction",
+        preferredID = "NotesViewTopComponent")
+@Messages({
+    "CTL_NotesViewAction=Notes View",
+    "CTL_NotesViewTopComponent=Notes View",
+    "HINT_NotesViewTopComponent=Notes View"})
+
+public class NotesViewTopComponent extends JavaFxTopComponent<NotesViewPane> implements GraphReportListener {
+    
+    private final NotesViewController notesViewController;
+    private final NotesViewPane notesViewPane;
+
+    /**
+     * NotesViewTopComponent constructor.
+     */
+    public NotesViewTopComponent() {
+        
+        setName(Bundle.CTL_NotesViewTopComponent());
+        setToolTipText(Bundle.HINT_NotesViewTopComponent());
+        
+        initComponents();
+        
+        notesViewController = new NotesViewController(this);
+        notesViewPane = new NotesViewPane(notesViewController);
+        
+        initContent();
+    }
+    
+    @Override
+    protected void handleNewGraph(final Graph graph) {
+
+        if (needsUpdate() && graph != null) {
+            notesViewPane.selectAllFilters();
+            notesViewPane.clearNotes();
+            notesViewPane.prepareNotesViewPane(notesViewController);
+        }
+    }
+
+    @Override
+    protected void handleGraphOpened(final Graph graph) {
+
+        if (needsUpdate() && graph != null) {
+            notesViewPane.selectAllFilters();
+            notesViewPane.clearNotes();
+            notesViewPane.prepareNotesViewPane(notesViewController);
+        }
+    }
+
+    @Override
+    protected void handleGraphClosed(final Graph graph) {
+
+        if (needsUpdate() && graph != null) {
+            notesViewPane.closeEdit();
+            notesViewPane.clearNotes();
+            notesViewPane.prepareNotesViewPane(notesViewController);
+        }
+    }
+
+    @Override
+    protected void handleComponentOpened() {
+        
+        GraphReportManager.addGraphReportListener(this);
+        final Graph activeGraph = GraphManager.getDefault().getActiveGraph();
+        
+        if (activeGraph != null) {
+            // Thread.sleep(100) used to ensure the following methods run in this order.
+            try {
+                Thread.sleep(100);
+                notesViewPane.clearNotes();
+                Thread.sleep(100);
+                notesViewPane.prepareNotesViewPane(notesViewController);
+                Thread.sleep(100);
+                notesViewPane.setGraphReport(notesViewController); // Plugin reports created while the Notes View is not "open" will appear when it is opened later.
+            } catch (final InterruptedException ex) {
+                Exceptions.printStackTrace(ex);
```

Comment:
Add a `Thread.currentThread().interrupt()` here given this is an Interrupted exception